core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/components/dot-uve-toolbar/dot-uve-toolbar.component.ts

```diff
@@ -82,31 +91,59 @@ export class DotUveToolbarComponent {
         return 'uve-toolbar uve-toolbar-preview';
     });
 
+    protected readonly publishDateParam = this.#store.pageParams().publishDate;
+    protected readonly $previewDate = model<Date>(
+        this.publishDateParam ? new Date(this.publishDateParam) : null
+    );
+
+    readonly $previewDateEffect = effect(
+        () => {
+            const previewDate = this.$previewDate();
+
+            if (!previewDate) {
+                return;
+            }
+
+            // If previewDate is minor that the CURRENT DATE, set previewDate to CURRENT DATE
+            if (previewDate < this.CURRENT_DATE) {
+                this.$previewDate.set(this.CURRENT_DATE);
+
+                return;
+            }
+
+            untracked(() => {
+                this.#store.loadPageAsset({
+                    editorMode: EDITOR_MODE.PREVIEW,
+                    publishDate: previewDate?.toISOString()
+                });
+            });
+        },
+        { allowSignalWrites: true }
+    );
```

Comment:
Why do we need this? Cannot we do this inside `triggerPreviewMode`

Reply:
because this effect allows filtering using the queryParams

Reply:
I think you should solve this in the shell, where we pass down the queryParams to the store

Reply:
Yes I set the value in the shell, but if the user enters directly through the link, the `triggerPreviewMode` function is not executed

and this effect is, since it is executed the first time the component is loaded (with the value of the queryParam, in this case)

Also with this effect i can react directly to the model changes (when user clicks on any date), I could react to the (change) event, put everything in `triggerPreviewMode` and call it in the onInit, but I think it would be the same result